packages/web-config-server/src/apiV1/utils/findAccessibleGroupedMapOverlays.js

```diff
@@ -91,14 +93,25 @@ const checkIfGroupedMapOverlaysAreEmpty = nestedMapOverlayGroups => {
   });
 };
 
+const allRelationsHaveSortOrder = relations =>
+  relations.every(m => m.sort_order !== undefined && m.sort_order !== null);
+
+const getSortFunction = relations => {
+  const childIdToSortOrder = reduceToDictionary(relations, 'child_id', 'sort_order');
+  const sortBySortOrder = (m1, m2) => childIdToSortOrder[m1.id] - childIdToSortOrder[m2.id];
+
+  //If all the relations have sort_order, sort by the sort_order. Otherwise by default sort by name.
+  return allRelationsHaveSortOrder(relations) ? sortBySortOrder : getSortByKey('name');
```

Comment:
I think my reasoning for either sort all by name or by orders (if all relations have sort order) is to encourage having sort orders for all map overlay items. In case that a dev missed an item when writing migration, I think it will be easier to spot. (I can't think of any cases that we actually want to sort only a portion of the items...?)

But if we want to be flexible then I think your suggested approach is certainly better.

Reply:
@chris-bes That also makes sense! Let's leave a comment in the original PR so that business will now, and let's keep the existing functionality 🙌